packages/core-typings/src/IUser.ts

```diff
@@ -79,27 +101,37 @@ export interface IUserServices {
 		changedAt: Date;
 	};
 	emailCode?: IUserEmailCode;
-	saml?: {
-		inResponseTo?: string;
-		provider?: string;
-		idp?: string;
-		idpSession?: string;
-		nameID?: string;
-	};
-	ldap?: {
-		id: string;
-		idAttribute?: string;
-	};
-	nextcloud?: {
-		accessToken: string;
-		refreshToken: string;
-		serverURL: string;
-	};
-	dolphin?: {
-		NickName?: string;
-	};
 }
 
+type IUserService = keyof IUserServices;
+type IOAuthServices = keyof IOAuthUserServices;
+
+const defaultOAuthKeys = [
+	'google',
+	'dolphin',
+	'facebook',
+	'github',
+	'gitlab',
+	'google',
+	'ldap',
+	'linkedin',
+	'nextcloud',
+	'saml',
+	'twitter',
+] as IOAuthServices[];
+const userServiceKeys = ['emailCode', 'email2fa', 'totp', 'resume', 'password', 'passwordHistory', 'cloud', 'email'] as IUserService[];
+
+export const isUserServiceKey = (key: string): key is IUserService => key in userServiceKeys || key in defaultOAuthKeys;
+
+export const isDefaultOAuthUser = (user: IUser): boolean =>
+	!!(user.services && Object.keys(user.services).some((key) => key in defaultOAuthKeys));
```

Comment:
```suggestion
	!!(user.services && Object.keys(user.services).some((key) => defaultOAuthKeys.includes(key)));
```
I believe `in` doesn't work to check if an array contains an element, does it? We should use `includes` instead